app/controllers/admin/content/events.js

```diff
@@ -0,0 +1,95 @@
+import Controller from '@ember/controller';
+
+export default Controller.extend({
+  actions: {
+    async loadSubTopics(topic) {
+      this.set('isLoading', true);
+      try {
+        this.set('subTopics', await topic.get('subTopics'));
+      } catch (e) {
+        this.notify.error(this.get('l10n').t('An unexpected error has occurred. SubTopics not loaded.'));
+      } finally {
+        this.set('isLoading', false);
+      }
+    },
+    openNewEventTypeModal(eventType) {
```

Comment:
The three can be,

```js
function openModalFor(modelInstanceOrName) {
  const modelName = typeof modelInstanceOrName === 'string' ? modelInstanceOrName : modelInstanceOrName.modelName;
  const modelInstance = typeof modelInstanceOrName === 'string' ? this.store.createRecord(modelInstanceOrName) : modelInstanceOrName;
  const camelCasedValue = camelCase(modelName); // from lodash
  this.set(camelCasedValue, modelInstance); // You can use this property itself to decide whether the model should be visible or not since `null` ==> falsy
}
```